### ISSUE TITLE
Do databases persist between different webkits?

### ISSUE THREAD

**Issue by kaansoral** [brody2consult/cordova-sqlite-storage#257]
I apologize if this question has been asked before or the answer is obvious

In the upcoming days, there will probably be a lot of webkit changes, so the localStorage isn't a persistent storage option anymore, when the webkit of the app changes, the data remains with the older one

Would using this plugin solve the persistence problem, does the database/location persist through different webkit utilisations?

Thanks in Advance

**Comment by nolanlawson** [comment]
This plugin saves data directly to SQLite via the native Objective-C/Java/C-sharp APIs. So no, none of the browser APIs like LocalStorage, WebSQL, or IndexedDB are used at all.

SQLite Plugin has a WebSQL-like API, but it is not WebSQL. :)

**Comment by kaansoral** [comment]
Thanks for the quick reply and confirmation

SQLite is an overkill for most scenarios, yet the cordova/browser solutions are not truly persistent and this plugin seems to be the best option for truly persistent storage, I will likely end up using one table, one row, one column with some serialized config and auth data :)

**Comment by nolanlawson** [comment]
LocalStorage, WebSQL, and IndexedDB are persistent. :) You can kill the app, reboot the phone, restart the app, and the data will still be there. (True, iOS does sometimes delete the data unexpectedly, but only if the user is running out of space.)

If you are using such a simple setup, might I recommend LocalStorage? `window.localStorage` can be used to store strings, and it's good enough, as long as you're storing <5MB of data.

**Comment by kaansoral** [comment]
It doesn't persist when you update the app to use a different webkit, for example, if cordova moves onto WKWebView, the localstorage data will likely stay at the UIWebView

Same for Android and Chrome/Crosswalk WebView's

The data also gets reset during updates, yet lately that's not the case

I wouldn't call this persistent :)

**Comment by nolanlawson** [comment]
Good point; now I see what you mean. This plugin would not have that problem. :)

**Comment by ammichael** [comment]
Hi there @kaansoral,

Sorry to resurrect the topic, but I've just this exactly problem right now. I was using WKWebview and PouchDB (WebSQL adapter) to store local data, truly believing it were reliable. Now, with iOS 11 the wkwebview version I was using started crashing, and when I updated it, the data weren't there.

I've opened an issue in the ionic repo's, but it seems you've been into this kinda of problem before. Is there any way to update the wkwebview keeping it's data?!

Thanks

**Comment by ttariq7** [comment]
@ammichael 
Experiencing the same issue with the WKWebView plugin. Was using 1.1.1-dev. Because of iOS 11 I have now upgraded to 1.1.4-dev but the localstorage is getting lost.

Did you find a solution?

**Comment by ammichael** [comment]
Hi @ttariq7 

The best solution I've found was to migrate the WKWebview data in iOS10 a new pouchdb instance usingSQLite plugin. When you upgrade to 1.1.4-dev your data will be kept properly.

The problem is all your users need to do the upgrade to migrate the data to the pouchdb using sqlite. If they skip the update and go straight to iOS 11, they will lose the data.

Hope it helps

**Comment by khatridev** [comment]
Had faced such issue while migrating from UIWebView to WKWebview. Used this fix to persist the local storage. Check if helps,

https://github.com/khatridev/cordova-plugin-migrate-localstorage

**Comment by brody2consult** [comment]
Hello everyone, the issue between webkits is only if you use `window.openDatabase`. I have already documented that you have to use `sqlitePlugin.openDatabase` to create and open a database with this plugin. And any database created with this plugin will be there when migrating from UIWebView to WKWebView.
